tests/test_group_by.cc

```diff
@@ -262,7 +263,7 @@ TYPED_TEST_P(GroupByTester, test_groupby_modify_combine) {
   const auto grouped = group_by(parent, this->test_case.get_grouper());
   auto groups = grouped.groups();
 
-  const auto first_key = map_keys(groups)[0];
+  const typename decltype(groups)::key_t first_key = map_keys(groups)[0];
```

Comment:
curious why `auto` doesn't work here?

Reply:
It deduces a reference type which has lifetime issues, triggering the address sanitizer!